R/infill_crits_interface.R

```diff
@@ -0,0 +1,366 @@
+#'      value by considered by \pkg{mlrMBO}?}
+#'   }
+#' @param name [\code{character(1)}]\cr
+#'   Full name of the criterion.
+#' @param id [\code{character(1)}]\cr
+#'   Short name of the criterion.
+#'   Used internally and in plots.
+#' @param minimize [\code{character(1)}]\cr
+#'   Shall the criterion be minimized or maximized?
+#'   Default is \code{TRUE}.
+#' @param components [\code{character}]\cr
+#'   Infill criteria may not return proposed point(s) only. Additional
+#'   information can be returned by appending a named \code{list} \dQuote{crit.components}
+#'   to the returned value as an attribute.
+#'   The \code{components} argument takes a character vector of the names of the
+#'   meta information, i.e., the names of the named \dQuote{crit.components} list.
+#'   Default is the empty character vector.
+#' @return [\code{MBOInfillCriterion}]
+#' @export
+makeMBOInfillCriterion = function(
+  fun,
+  name,
+  id,
+  minimize = TRUE,
+  components = character(0L)) {
+  assertFunction(
+    fun,
+    args = c("points", "models", "control",
+      "par.set", "design", "iter", "attributes"),
+    ordered = TRUE)
+
+  assertString(name)
+  assertString(id)
+  assertFlag(minimize)
+  assertCharacter(components, unique = TRUE)
+
+  ic = makeS3Obj("MBOInfillCriterion",
+    fun = fun,
+    name = name,
+    id = id,
+    minimize = minimize,
+    components = components
+  )
+  return(ic)
+}
+
+getMBOInfillCritParams = function(x) {
+  assertClass(x, "MBOInfillCriterion")
+  return(x$params)
+}
+
+getMBOInfillCritParam = function(x, par.name) {
+  assertClass(x, "MBOInfillCriterion")
+  assertString(par.name)
+  return(getMBOInfillCritParams(x)[[par.name]])
+}
+
+getMBOInfillCritName = function(x) {
+  assertClass(x, "MBOInfillCriterion")
+  return(x$name)
+}
+
+getMBOInfillCritId = function(x) {
+  assertClass(x, "MBOInfillCriterion")
+  return(x$id)
+}
+
+getMBOInfillCritId = function(x) {
+  assertClass(x, "MBOInfillCriterion")
+  return(x$id)
+}
+
+getMBOInfillCritComponents = function(x) {
+  assertClass(x, "MBOInfillCriterion")
+  return(x$components)
+}
+
+getMBOInfillDummyCritComponents = function(x) {
+  assertClass(x, "MBOInfillCriterion")
+  ns = getMBOInfillCritComponents(x)
+  as.data.frame(BBmisc::namedList(ns, NA_real_))
+}
+
+getMBOInfillCritDirection = function(x) {
+  assertClass(x, "MBOInfillCriterion")
+  ifelse(x$minimize, 1, -1)
+}
+
+#' @title Infill criteria.
+#' @name infillcrits
+#'
+#' @description
+#' \pkg{mlrMBO} contains most of the most popular infill criteria.
+#'
+#' @param cb.lambda [\code{numeric(1)}]\cr
+#'   Lambda parameter for confidence bound infill criterion.
+#'   In the multi-objective case we recommend to set this value to
+#'   \eqn{q(0.5 \cdot \pi_{CB}^(1 / n))} where \eqn{q} is the quantile
+#'   function of the standard normal distribution, \eqn{\pi_CB} is the probability
+#'   of improvement value and \eqn{n} is the number of objectives of the considered problem.
+#'   Default is 1.
```

Comment:
Tobias formula \eqn{q(0.5 \cdot \pi_{CB}^(1 / n))} is meant for the DIB infill criterion only.

I think this could be confusing for the "normal user" who does not know about the multi objective case. Perhaps use different names for the parameters? One for the cb, and one for the dib criterion. cb.lambda and dib.lambda?